eventcore-postgres/src/lib.rs

```diff
@@ -17,11 +18,35 @@ pub enum PostgresEventStoreError {
     ConnectionFailed(#[source] sqlx::Error),
 }
 
+/// Maximum number of database connections in the pool.
+///
+/// MaxConnections represents the connection pool size limit. It must be at least 1
+/// (enforced by NonZeroU32) and should not exceed reasonable database limits
+/// (typically 100-1000 depending on your database server configuration).
+///
+/// # Examples
+///
+/// ```ignore
+/// use eventcore_postgres::MaxConnections;
+///
+/// let small_pool = MaxConnections::try_new(5).unwrap();
+/// let standard = MaxConnections::try_new(10).unwrap();
+/// let large_pool = MaxConnections::try_new(50).unwrap();
+///
+/// // Zero connections not allowed
+/// assert!(MaxConnections::try_new(0).is_err());
+/// ```
+#[nutype(
+    validate(greater = 0, less_or_equal = 1000),
+    derive(Debug, Clone, Copy, PartialEq, Eq, Display, AsRef, Into, TryFrom)
```

Comment:
Requiring it to be greater than 0 makes sense, but lte 1000 seems completely arbitrary. Also, the comment above says it is enforced by using NonZeroU32 as the underlying type, but that's a lie; it is currently using u32. I think we should change it to NonZeroU32 and then we can drop both validations.